### PR TITLE
TRM: IP multicast provider, type and beaker testcases.

### PR DESCRIPTION
**Pull request by rahushen**

### CODE REVIEW

File: examples/cisco/demo_trm.pp [hunk 1]
```diff
@@ -49,6 +49,12 @@
         route_target_export_mvpn      => $rt_export_mvpn,
         route_target_both_auto_mvpn   => $rt_both_mvpn,
       }
+
+      cisco_ip_multicast { 'default':
+        ensure =>                 present,
+        overlay_distributed_dr => 'true',
+        overlay_spt_only =>       'true',
```
Comment (mikewiebe) [inline examples/cisco/demo_trm.pp]: Make sure to enable `nv overlay evpn` in the node_utils for `overlay_spt_only`

File: lib/puppet/type/cisco_ip_multicast.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  # Parameters #
+  ##############
+  newparam(:name, namevar: true) do
+    # Creating a parameter to satisfy namevar condition
+    # Only 'default' is an accepted value
+    desc "Instance of IP Multicast, only allow the value 'default'"
+
+    validate do |name|
+      if name != 'default'
+        error "only 'default' is accepted as a valid ip_multicast resource name"
```
Comment (mikewiebe) [inline lib/puppet/type/cisco_ip_multicast.rb]: *nit* I would prepend ip_multicast with cisco.
`error "only 'default' is accepted as a valid cisco_ip_multicast resource name"`

File: tests/beaker_tests/cisco_ip_multicast/test_ip_multicast.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+# Test hash test cases
+tests[:default] = {
+  desc:           '1.1 Default Properties',
+  title_pattern:  'default',
+  manifest_props: {
+    overlay_distributed_dr: 'default',
+    overlay_spt_only:       'default',
+  },
+  resource:       {}, # no output expected when all defaults
```
Comment (mikewiebe) [inline tests/beaker_tests/cisco_ip_multicast/test_ip_multicast.rb]: If the manifest uses `ensure => present` then I would expect the resource to display both properties set to the default value of `false`.

File: examples/cisco/demo_trm.pp (hunk 1 above)
Comment (rahushen) [reply examples/cisco/demo_trm.pp]: Done.

File: lib/puppet/type/cisco_ip_multicast.rb (hunk 2 above)
Comment (rahushen) [reply lib/puppet/type/cisco_ip_multicast.rb]: done

File: tests/beaker_tests/cisco_ip_multicast/test_ip_multicast.rb (hunk 3 above)
Comment (rahushen) [reply tests/beaker_tests/cisco_ip_multicast/test_ip_multicast.rb]: done

**Comment by mikewiebe** [comment]
👍  Will merge when travis completes.

**Pull request merged**